File: olympiads/1-10-2023/3.cpp
#include <iostream>
#include <vector>
#include <algorithm>

typedef long long ll;

const ll MOD = 998244353;

ll factorial(ll n) {
    ll result = 1;
    for (ll i = 2; i <= n; ++i) {
        result = (result * i) % MOD;
    }
    return result;
}

ll countPaths(std::vector<ll>& a1) {
    ll sum = 0;
    for (ll a : a1) {
        sum += a;
    }
    ll result = factorial(sum);
    ll divider = 1;
    for (ll a : a1) {
        divider = (divider * factorial(a)) % MOD;
    }
    if (divider != 0) {
        result = (result / divider) % MOD;
    }
    return result % MOD;
}

int main() {
    ll N;
    std::cin >> N;
    std::vector<ll> a1(N);
    for (ll i = 0; i < N; ++i) {
        std::cin >> a1[i];
    }
    ll result = 1;
    if (N == 1) {
        result = 1;
    } else {
        result = countPaths(a1);
    }
    std::cout << result << std::endl;
    return 0;
}

File: olympiads/1-10-2023/3.py
from math import factorial


MOD = 998244353


def countPaths(a1):
    sum = 0
    for a in a1:
        sum += a
    result = factorial(sum)
    divider = 1
    for a in a1:
        divider *= factorial(a)
    if divider != 0:
        result //= divider
    return (result % MOD)


N = int(input())
a1 = list(map(int, input().split()))
result = 1
if (N == 1):
    result = 1
else:
    result = countPaths(a1)
print(result)